### DISCUSSION TITLE
[Chapter 08] @Retryable

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/spring-boot-for-msa#130] [Chapter 08. Data Persistence] [answered]
Q1. 

p. 578 
>   ~의 메서드가 실패할 수도 있다.  이때는  `@Retryable`  애너테이션 기능을 사용하여 재시도하는 방법을 사용한다. 혹은 신뢰성이 높은 메시지 큐를 고려하는 것도 바람직하다.

책에서 MSA 환경에서 데이터를 처리하는 작업과 다른 서버와 협업하는 작업 모두를 100% 동기화하기는 쉽지 않다. 라는 설명이 나오면서 위와 같은 설명이 되어있다. 
처음 보는 애너테이션인데 책에 스쳐지나가듯이 설명이 되어있어서 찾아봤는데 꽤나 유용하게 사용될 거 같아 궁금해졌다. 
혹시 프로젝트나 실제 업무에서 이것을 사용해 본 적이 있을까? 
그리고 실제 MSA 환경에서 이런

**Comment by JoisFe** [comment]
kafka를 사용중에 있고 consumer retry max-attempts 3회 두고있습니다.

**Comment by Irisation23** [comment, marked as answer]
<img width="912" alt="image" src="https://github.com/cowaine/spring-boot-for-msa/assets/53285909/5aba3f65-8be4-4648-8aac-44849efa282a">

의 내용을 볼 수 있었네요~

> 출처
> > https://yeon-kr.tistory.com/213
